src/libpmemfile-core/file.c

```diff
@@ -1674,3 +1674,158 @@ pmemfile_truncate(PMEMfilepool *pfp, const char *path, off_t length)
 
 	return 0;
 }
+
+static int
+vinode_chown(PMEMfilepool *pfp, struct pmemfile_cred *cred,
+		struct pmemfile_vinode *vinode, uid_t owner, gid_t group)
+{
+	struct pmemfile_inode *inode = vinode->inode;
+	int error = 0;
+
+	os_rwlock_wrlock(&vinode->rwlock);
+
+	TX_BEGIN_CB(pfp->pop, cb_queue, pfp) {
+		if (!(cred->caps & (1 << PMEMFILE_CAP_CHOWN))) {
```

Comment:
These ten lines can be moved, to be outside the transaction. Also, probably there would be no need to split the 'if (owner != -1 && owner != cred->fsuid)' line into two lines, if it had one fewer tab in front of it.